### ISSUE TITLE
Yaml exception when trying to install rdoc-data

### ISSUE THREAD

**Issue by bo-tato** [jruby/jruby#7939]
I get:
```
Exception while verifying /home/user/.rvm/rubies/jruby-head/lib/ruby/gems/shared/cache/rdoc-data-4.1.0.gem
ERROR:  While executing gem ... (Java::OrgSnakeyamlEngineV2Exceptions::YamlEngineException)
    The incoming YAML document exceeds the limit: 3145728 code points.
```
when running `rvm docs generate`, which in turn is running `gem install rdoc-data` when it gets the yaml error. This happens with the latest jruby release in RVM (jruby-9.4.3.0) and installing jruby-head (git master) with RVM.

**Comment by bo-tato** [comment]
Thanks, I think it needs to be changed in the rubygems code that comes with jruby. It's not specific to rdoc-data, in fact trying to install other gems after is broken until I delete rdoc-data-4.1.0.gem from the cache as gem still tries and fails to parse it's yaml. It seems rubygems is calling Psych.safe_load, which uses the default parser and doesn't have a way to pass a custom parser with that code_point_limit set, so I tried just changing the value for the default parser. In `jruby-9.4.3.0/lib/ruby/stdlib/rubygems.rb` in `self.load_yaml` after `require 'psych'` I added `Psych.parser.code_point_limit = 20_000_000` but it doesn't have an effect, I still get a crash with code point limit of 3145728. I'm also confused as the code_point_limit attribute exists on Psych parser, but `Psych::Parser.instance_method(:code_point_limit).source_location` returns nil, and grepping the source for jruby I can't find any reference to it. There is a psych.jar file bundled in with jruby, maybe it is in there?

**Comment by headius** [comment]
The ability to set this globally was merged into Psych in https://github.com/ruby/psych/pull/647 but not yet released.

**Comment by headius** [comment]
I've pushed an additional PR for Psych that will remove three ineffective settings (https://github.com/ruby/psych/pull/653). Once that is merged I believe we can get a release of Psych out that exposes the code point limit as a global setting.

**Comment by headius** [comment]
I have merged https://github.com/ruby/psych/pull/653 and once we get a release of Psych we can include it in JRuby 9.4.4.0.

**Comment by headius** [comment]
Fixed by the update in #7966.
